demo/admin/src/products/future/generated/ProductsGrid.tsx

```diff
@@ -99,7 +101,34 @@ export function ProductsGrid({ filter, toolbarAction, rowAction }: Props): React
     const dataGridProps = { ...useDataGridRemote(), ...usePersistentColumnState("ProductsGrid") };
 
     const columns: GridColDef<GQLProductsGridFutureFragment>[] = [
-        { field: "inStock", headerName: intl.formatMessage({ id: "product.inStock", defaultMessage: "In stock" }), type: "boolean", width: 90 },
+        {
+            field: "inStock",
+            headerName: intl.formatMessage({ id: "product.inStock", defaultMessage: "In Stock" }),
+            type: "singleSelect",
+            valueOptions: [
+                { value: "true", label: intl.formatMessage({ id: "product.inStock.true.primary", defaultMessage: "In stock" }) },
+                { value: "false", label: intl.formatMessage({ id: "product.inStock.false.primary", defaultMessage: "Out of stock" }) },
+            ],
+            renderCell: ({ row }) => {
+                const valueLabels: Record<string, React.ReactNode> = {
```

Comment:
Can we extend the existing valueOptions to include the react node?

maybe like so:

`{ value: "true", label: ..., renderCell: () => <GridCellContent
                            primaryText={<FormattedMessage id="product.inStock.true.primary" defaultMessage="In stock" />}
                            icon={<StateFilledIcon color="success" />}
                        /> },`

Reply:
We could do something like this: 

```tsx
const inStockValueOptions = [
    {
        value: "true",
        label: intl.formatMessage({ id: "product.inStock.true.primary", defaultMessage: "In stock" }),
        node: (
            <GridCellContent
                primaryText={<FormattedMessage id="product.inStock.true.primary" defaultMessage="In stock" />}
                icon={<StateFilledIcon color="success" />}
            />
        ),
    },
    {
        value: "false",
        label: intl.formatMessage({ id: "product.inStock.false.primary", defaultMessage: "Out of stock" }),
        node: (
            <GridCellContent
                primaryText={<FormattedMessage id="product.inStock.false.primary" defaultMessage="Out of stock" />}
                icon={<StateFilledIcon color="error" />}
            />
        ),
    },
];
// ...
{
    field: "inStock",
    valueOptions: inStockValueOptions,
    renderCell: ({ value }) => {
        const renderCellValue = inStockValueOptions.find((option) => option.value === value.toString());
        return renderCellValue ? renderCellValue.node : value.toString();
    },
}
```

Would you prefer this? 
I don't really see the benefit in combining this into a variable, as the only thing they share is the `value` 🤔

Reply:
see show I implemented it inthis  WIP PR: https://github.com/vivid-planet/comet/pull/2276/files#diff-83d277e6e17812c36590b99d6b654e727c2b0734bfa2ef70901793b989f65872R124

(that has value/label only, but also shared for filter+renderCell)

Reply:
> as it requires changing the type of valueOptions to support an additional renderCell/cellContent key and also requires a bunch of additional type-checks.

if the render function comes from the library the additional type-checks are no problem

Reply:
So you actually like this? 😅 1d6dbfafafdbc42d992d7aada5a222cc9683a255

Reply:
yes! why not?

Reply:
`cellContent` just seems to be forced into the `valueOptions` in a way I would not do, for non-generated code 😄  
But maybe it's fine if we add a function to the library to abstract away all the ifs/type-checks.

Reply:
I'm with @jamesricky on this. I wouldn't write code like this. But maybe we need to let go of this guideline?

Reply:
What do you think of the example I mentioned above https://github.com/vivid-planet/comet/pull/2422#discussion_r1724686784, with the separate variable for the `valueOptions`, defined above all the column definitions? 

We would have to do `.map(({ value, label }) => ({ value, label }))` to apply it to the `GridColDef` but could access the `node`/`renderCell` using the variable inside the `renderCell` function.

Reply:
> What do you think of the example I mentioned above https://github.com/vivid-planet/comet/pull/2422#discussion_r1724686784, with the separate variable for the valueOptions, defined above all the column definitions?

What would be the benefit of this? Getting rid of the if and type checks in 1d6dbfafafdbc42d992d7aada5a222cc9683a255? Otherwise it seems pretty similar to me...

Reply:
> What would be the benefit of this? Getting rid of the if and type checks in https://github.com/vivid-planet/comet/commit/1d6dbfafafdbc42d992d7aada5a222cc9683a255?

Yes, exactly that 😄

Reply:
I'm not convinced. Defining the options somewhere else while they can be retrieved from the column def seems impractical. However, for handmade code, I'd probably do exactly that 😅

Reply:
Done in 9c69a2f3cd779b37e99c14f735265c7cd94d0ca8 and 93caea33416a8ceef7d95cc78ded54398cdd4a28